modules/ossm-threescale-webassembly-module-examples-for-credentials-use-cases.adoc

```diff
@@ -0,0 +1,236 @@
+
+`aladdin:opensesame`:  Authorization: Basic YWxhZGRpbjpvcGVuc2VzYW1l
+
+You must use lower case header field names as shown in the following example:
+
+[source,yaml]
+----
+credentials:
+  app_id:
+    - header:
+        keys:
+          - authorization
+        ops:
+          - split:
+              separator: " "
+              max: 2
+          - length:
+              min: 2
+          - drop:
+              head: 1
+          - base64_urlsafe
+          - split:
+              max: 2
+  app_key:
+    - header:
+        keys:
+          - app_key
+----
+
+The previous example use case looks at the headers for an `authorization`:
+
+. It takes its string value and split it by a space, checking that it generates at least two values of a `credential`-type and the `credential` itself, then dropping the `credential`-type.
+. It then decodes the second value containing the data it needs, and splits it by using a colon (:) character to have an operations stack including first the `app_id`, then the `app_key`, if it exists.
+.. If `app_key` does not exist in the authorization header then its specific sources are checked, for example, the header with the key `app_key` in this case.
++
+You can always add extra conditions to `credentials`, for example:
++
+. Ensure to only allow `Basic` authorizations, where `app_id` is either `aladdin` or `admin`, or any `app_id` being at least 8 characters in length.
+. `app_key` must contain a value and have a minimum of 64 characters as shown in the following example:
++
+[source,yaml]
+----
+credentials:
+  app_id:
+    - header:
+        keys:
+          - authorization
+        ops:
+          - split:
+              separator: " "
+              max: 2
+          - length:
+              min: 2
+          - reverse
+          - glob:
+            - Basic
+          - drop:
+              tail: 1
+          - base64_urlsafe
+          - split:
+              max: 2
+          - test:
+              if:
+                length:
+                  min: 2
+              then:
+                - strlen:
+                    max: 63
+                - or:
+                    - strlen:
+                        min: 1
+                    - drop:
+                        tail: 1
+          - assert:
+            - and:
+              - reverse
+              - or:
+                - strlen:
+                    min: 8
+                - glob:
+                  - aladdin
+                  - admin
+----
++
+. After picking up the `authorization` header value, you get a `Basic` `credential`-type by reversing the stack so that the type is placed on top.
+. Run a glob match on it. When it validates, and the credential is decoded and split, you get the `app_id` at the bottom of the stack, and potentially the `app_key` at the top.
+. Run a `test:` if there are two values in the stack, meaning we got an `app_key`.
```

Comment:
Suggest avoiding "we". Maybe just "meaning an `app_key` was acquired" or something.